apps/api/src/passport/test/e2e.spec.ts

```diff
@@ -895,6 +901,92 @@ describe("E2E Tests", () => {
       });
     });
   });
+
+  describe("Login attempts", () => {
+    beforeEach(async () => {
+      const module = await Test.createTestingModule(moduleMetaData).compile();
+
+      req = module.get(Request);
+      app = module.createNestApplication();
+
+      await app.listen(req.socket);
+
+      // WAIT UNTIL IDENTITY IS INSERTED
+      await new Promise((resolve, _) => setTimeout(resolve, 3000));
+
+      jest.useFakeTimers({doNotFake: ["nextTick"]});
+    });
+
+    afterEach(() => {
+      jest.useRealTimers();
+    });
+
+    it("should block if the attempt limit is exceeded ", async () => {
+      const responses = [];
+      for (const fn of Array(3).fill(() => login("spica", "wrongPassword"))) {
+        responses.push(await fn());
+      }
+
+      const responseWithBlockedError = responses[responses.length - 1];
+
+      expect(responseWithBlockedError.statusCode).toEqual(401);
+      expect(responseWithBlockedError.statusText).toEqual("Unauthorized");
+
+      const expectedMessages = [
+        "Too many failed login attempts. Try again after 10 minutes.",
+        "Too many failed login attempts. Try again after 9 minutes 59 seconds."
```

Comment:
They could be failed if tests take more than two seconds, instead of this, you can use jest.setSystemTime to fix the time.